docs: added release notes to changelog.md (#569)
Unity-Technologies/com.unity.multiplayer.mlapi: com.unity.multiplayer.mlapi/CHANGELOG.md

- GitHub [444](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/444) and [455](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/455): Channels are now represented as bytes instead of strings.

For users of previous versions of MLAPI, this release renames APIs due to refactoring. All obsolete marked APIs have been removed as per [GitHub 513](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/513) and [GitHub 514](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/514).

| Previous MLAPI Versions | V 0.1.0 Name |
| -- | -- |
| `NetworkingManager` | `NetworkManager` |
| `NetworkedObject` | `NetworkObject` |
| `NetworkedBehaviour` | `NetworkBehaviour` |
| `NetworkedClient` | `NetworkClient` |
| `NetworkedPrefab` | `NetworkPrefab` |
| `NetworkedVar` | `NetworkVariable` |
| `NetworkedTransform` | `NetworkTransform` |
| `NetworkedAnimator` | `NetworkAnimator` |
| `NetworkedAnimatorEditor` | `NetworkAnimatorEditor` |
| `NetworkedNavMeshAgent` | `NetworkNavMeshAgent` |
| `SpawnManager` | `NetworkSpawnManager` |

> | `BitStream/BitSerializer`... | `NetworkBuffer/NetworkSerializer`... |

Reviewer: ```suggestion
| `BitStream` | `NetworkBuffer`|
| `BitWriter` | `NetworkWriter`|
| `BitReader` | `NetworkReader`|
```

Reviewer: @MFatihMAR NetworkSerializer/BitSerializer hasn't been part of the last release so not putting it into the changelog. Does that make sense?

Reviewer 2: yeah, this is not a name change. this is a new code/feature we implemented.
I still think it should be in the changelog though, perhaps not under renaming but on the new features added section.

Reviewer 2: @lkrell

Author: On it thanks!

| `NetEventType` | `NetworkEventType` |
| `ChannelType` | `NetworkDelivery` |
| `Channel` | `NetworkChannel` |
| `Transport` | `NetworkTransport` |
| `NetworkedDictionary` | `NetworkDictionary` |
| `NetworkedList` | `NetworkList` |
| `NetworkedSet` | `NetworkSet` |

> | `MLAPIConstants` | `NetworkConstants` |

Reviewer: ```suggestion
| `MLAPIConstants` | `NetworkConstants` |
| `UnetTransport` | `UNetTransport` |
```

# Changelog
This file documents all notable changes to this package. Additional documentation and release notes are available at [Multiplayer Documentation](https://docs-multiplayer.unity3d.com).

## [0.1.0] - 2021-03-23

This is the initial experimental Unity MLAPI Package, v0.1.0.

### New Features

> - Refactored a new standard for Remote Procedure Call (RPC) in MLAPI which provides increased performance, significantly reduced boilerplate code, and extensibility for future-proofed code. MLAPI RPC includes `ServerRpc` and `ClientRpc` to execute logic on the server and client-side.

Reviewer 3: I actually think for user clarity, and since this is a changelog after all, the verbiage explaining how we formerly had perf / conv RPCs and now just have a single performant unified RPC solution should remain.

Author: I have info in the removed features on removed conv/perf, can note it in this entry with a slight tweak so it's not redundant :)

- Added standarized serialization types, including built-in and custom serialization flows. See [RFC #2](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/master/text/0002-serializable-types.md) for details.
- `INetworkSerializable` interface replaces `IBitWritable`.

> - Added `NetworkSerializer`... (was `BitSerializer`...), which is the main aggregator that implements serialization code for built-in supported types and holds `BitReader` and `BitWriter` instances internally.

Reviewer 2: ```suggestion
- Added `NetworkSerializer` which is the main aggregator that implements serialization code for built-in supported types and holds `NetworkReader` and `NetworkWriter` instances internally.
```

Reviewer 2: people never saw `BitSerializer` before, no need to mention over here :P

- Added a Network Update Loop infrastructure that aids Netcode systems to update (such as RPC queue and transport) outside of the standard `MonoBehaviour` event cycle. See [RFC #8](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/master/text/0008-network-update-loop.md) and the following details:

  - It uses Unity's [low-level Player Loop API](https://docs.unity3d.com/ScriptReference/LowLevel.PlayerLoop.html) and allows for registering `INetworkUpdateSystem`s with `NetworkUpdate` methods to be executed at specific `NetworkUpdateStage`s, which may also be before or after `MonoBehaviour`-driven game logic execution.
  - You will typically interact with `NetworkUpdateLoop` for registration and `INetworkUpdateSystem` for implementation.
  - `NetworkVariable`s are now tick-based using the `NetworkTickSystem`, tracking time through network interactions and syncs.

> - Added message batching to handle consecutive RPC requests sent to the same client. `MessageBatcher` sends batches based on requests from the `RpcQueueProcessing`, by batch size threshold or immediately.

Reviewer 2: ```suggestion
- Added message batching to handle consecutive RPC requests sent to the same client. `RpcBatcher` sends batches based on requests from the `RpcQueueProcessing`, by batch size threshold or immediately.
```

It's called `RpcBatcher` not `MessageBatcher` now :)

- [GitHub 494](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/494): Added a constraint to allow one `NetworkObject` per `GameObject`, set through the `DisallowMultipleComponent` attribute.
- Integrated MLAPI with the Unity Profiler for versions 2020.2 and later.

> - A test project is available for building and experimenting with MLAPI features. This project is located in the package `testproject` folder ([GitHub folder](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/tree/release/0.1.0/testproject)). 

Reviewer 3: How does this changelog show up for package users?  Reason I ask is, for them they will not have the testproject

Author: Hmm can revise pointing them to the test project on github, adding a tweak!

### Removed features

With a new release of MLAPI in Unity, some features have been removed:

* SyncVars have been removed from MLAPI. Use `NetworkVariable`s in place of this functionality. <!-- MTT54 -->
* [GitHub 527](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/527): Lag compensation systems and `TrackedObject` have moved to the new [MLAPI Community Contributions](https://github.com/Unity-Technologies/mlapi-community-contributions/tree/master/com.mlapi.contrib.extensions) repo.
* [GitHub 509](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/509): Encryption has been removed from MLAPI. The `Encryption` option in `NetworkConfig` on the `NetworkingManager` is not available in this release. This change will not block game creation or running. A current replacement for this functionality is not available, and may be developed in future releases. See the following changes:

    * Removed `SecuritySendFlags` from all APIs.

>     * Revmoed encryption, cryptography, and certificate configurationss from APIs including `NetworkManager` and `NetworkConfig`.

Reviewer 2: ```suggestion
    * Removed encryption, cryptography, and certificate configurations from APIs including `NetworkManager` and `NetworkConfig`.
```

>     * Removed "hail handshake", including `NetworkManager` implementation and `MLAPIConstants` entries.

Reviewer 2: ```suggestion
    * Removed "hail handshake", including `NetworkManager` implementation and `NetworkConstants` entries.
```

- Refactored a new standard for Remote Procedure Call (RPC) in MLAPI which provides increased performance, significantly reduced boilerplate code, and extensibility for future-proofed code. MLAPI RPC includes `ServerRpc` and `ClientRpc` to execute logic on the server and client-side.
- Added standarized serialization types, including built-in and custom serialization flows. See [RFC #2](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/master/text/0002-serializable-types.md) for details.
- `INetworkSerializable` interface replaces `IBitWritable`.

> - Added `NetworkSerializer`... and `BitSerializer`..., which is the main aggregator that implements serialization code for built-in supported types and holds `BitReader` and `BitWriter` instances internally.

Reviewer 2: nope, `NetworkSerializer` is the `BitSerializer` with a different name. we initially named it `BitSerializer` but changed it to `NetworkSerializer`. so basically, we're only adding `NetworkSerializer` :) (I know it was confusing, sorry!)

Author: All good! editing!

Reviewer 2: also please replace `BitReader` and `BitWriter` with `NetworkReader` and `NetworkWriter`.

| `NetworkedAnimatorEditor` | `NetworkAnimatorEditor` |
| `NetworkedNavMeshAgent` | `NetworkNavMeshAgent` |
| `SpawnManager` | `NetworkSpawnManager` |

> | `BitStream` | `NetworkBuffer` |

Reviewer 2: I'd suggest listing these:
- `BitStream` → `NetworkBuffer`
- `BitReader` → `NetworkReader`
- `BitWriter` → `NetworkWriter`
